test/e2e/memcached_test.go

```diff
@@ -68,7 +68,47 @@ func TestMemcached(t *testing.T) {
 		"new",
 		"memcached-operator").CombinedOutput()
 	if err != nil {
-		t.Fatalf("error: %v\nCommand Output: %s\n", err, string(cmdOut))
+		// HACK: dep cannot resolve non-master branches as the base branch for PR's,
+		// so running `dep ensure` will fail when first running
+		// `operator-sdk new ...`. For now we can ignore the first solve failure,
+		// and run `dep ensure` again.
+		// A permanent solution can be implemented once the following is merged:
+		// https://github.com/golang/dep/pull/1658
+		solveFailRe := regexp.MustCompile(`(?m)^[ \t]*Solving failure:.+github\.com/operator-framework/operator-sdk.+:$`)
+		if solveFailRe.Match(cmdOut) {
```

Comment:
@estroz I think we always need to update the Gopkg.toml Source field for the SDK to the contributor's PR branch(when running in travis on a PR).

This shouldn't be conditional on `operator-sdk new` failing.
We just need to ignore the specific `Solving failure: ...` error for `operator-sdk new` and proceed with the rest of the test.
Otherwise if `operator-sdk` doesn't fail we won't change the Source to the PR branch.
/cc @AlexNPavel

Reply:
Yep I'm reverting that now